rename: addition

mkgvt opened the pull request:
###### Motivation for this change

Package alternative rename command which allows file renames to be specified via a perl regex.

###### Things done

Package has been built and installed by hand in user environment via nix-build/nix-env.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Mic92 committed:
rancher-compose: set version during build

mkgardner committed:
Merge remote-tracking branch 'upstream/master'

mkgardner committed:
Merge remote-tracking branch 'upstream/master'

mkgardner committed:
Merge remote-tracking branch 'upstream/master'

mkgardner committed:
rename: init at 1.9

rycee commented:
Since this is a utility that just happens to be implemented in Perl so I think it is better put in, e.g., `pkgs/tools/misc/rename/default.nix` like any other package. You can have a look at `./pkgs/tools/typesetting/biber/default.nix` for an example of a "standalone Perl package".

rycee commented on pkgs/top-level/perl-packages.nix:
```diff
@@ -11884,6 +11884,20 @@ let self = _self // overrides; _self = with self; {
+      description = "Bulk rename files according to a Perl rewrite expression";
+      homepage = http://search.cpan.org/~pederst/rename-1.9/bin/rename.PL;
+      maintainers = with maintainers; [ mkg ];
+      license = "unknown";
```
This should be

    license = with stdenv.lib.licenses; [ artistic1 gpl1Plus ];

according to the [LICENSE](https://github.com/pstray/rename/blob/master/LICENSE) file in the repo.

mkgardner committed:
Merge remote-tracking branch 'upstream/master'

mkgardner committed:
rename: incorporated PR reviews and submission process from manual
  - moved from perl-packages to pkgs/tools/misc/rename
  - corrected license
  - switched to fetching from github

Mic92 committed:
rename: nitpicks
